moxin-frontend/src/my_models/models_table.rs

```diff
@@ -0,0 +1,420 @@
+                        color: #FFF,
+                        border_color: #B4B4B4,
+                        border_width: 1.0,
+                        instance radius: 2.0,
+                    }
+                    attr_name = {
+                        draw_text: {
+                            text_style: <BOLD_FONT>{font_size: 9}
+                            color: #x0
+                        }
+                    }
+                }
+            }
+            file_size_tag = <View> {
+                width: 80
+                align: {x: 0.0, y: 0.5}
+                file_size = <Label> {
+                    draw_text: {
+                        text_style: <REGULAR_FONT>{font_size: 9}
+                        color: #x0
+                    }
+                }
+            }
+            compatibility_guess_tag = <View> {
+                width: 130
+                align: {x: 0.0, y: 0.5}
+                compatibility = <AttributeTag> {
+                    draw_bg: { color: #E6F1EC },
+                    attr_name = { draw_text: { color: #101828} }
+                }
+            }
+            date_added_tag = <View> {
+                width: 80
+                align: {x: 0.0, y: 0.5}
+                date_added = <Label> {
+                    draw_text: {
+                        text_style: <REGULAR_FONT>{font_size: 9}
+                        color: #x0
+                    }
+                }
+            }
+
+            actions = <View> {
+                width: 80
+                flow: Right
+                spacing: 10
+                align: {x: 0.0, y: 0.5}
+
+                <ActionButton> { type_: Play, icon = { draw_icon: { svg_file: (ICON_PLAY) } } }
+                <ActionButton> { type_: Info, icon = { draw_icon: { svg_file: (ICON_INFO) fn get_color(self) -> vec4 { return #0099FF;} } } }
+                <ActionButton> { type_: Delete, icon = { draw_icon: { svg_file: (ICON_DELETE) fn get_color(self) -> vec4 { return #B42318;} } } }
+            }
+        }
+    }
+
+    ModelsTable = {{ModelsTable}} <RoundedView> {
+        width: Fill,
+        height: Fill,
+        align: {x: 0.5, y: 0.5}
+
+        list = <PortalList>{
+            drag_scrolling: false
+            HeaderRow = <HeaderRow> {}
+            ItemRow = <Row> {}
+        }
+    }
+}
+
+#[derive(Live, LiveHook, Widget)]
+pub struct ModelsTable {
+    #[deref]
+    view: View,
+    #[rust]
+    model_item_map: HashMap<u64, String>,
+}
+
+impl Widget for ModelsTable {
+    fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
+        let widget_uid = self.widget_uid();
+        for list_action in cx.capture_actions(|cx| self.view.handle_event(cx, event, scope)) {
+            if let Some(action) = list_action.as_widget_action() {
+                if let Some(group) = &action.group {
+                    match list_action.as_widget_action().cast() {
+                        RowAction::PlayClicked => {
```

Comment:
Already discussed in person, but looking at the code I'm more convinced that the version without using a custom widget for the button should be much simpler. You wouldn't need the `RowAction`.

```rs
if let Some(fe) = self.view(id!(play_button)).finger_up(&actions) {
    if fe.was_tap() { ... }
}
if let Some(fe) = self.view(id!(info_button)).finger_up(&actions) {
    if fe.was_tap() { ... }
}
...
```

However, I see my proposal is not enough because you need to identify which row was clicked. My alternative only work when you have an specific widget for the row tables, so you have there an scope where there is only one  instance of `play_button`, `info_button`, etc (which I think it is common way of designing tables/lists components where items has some behavior apart from displaying info).

I wouldn't change anything in this PR, but I'm not seeing a clear path to migrate to Makepad buttons from this point. But let's see.

Reply:
Fair point, I was avoiding creating widgets for each button.
I think we can rework this when swapping the `RoundedView`s for Makepad `Button`s
Although IIRC for other similar cases (WeChat and Taobao) I remember still needing a row-level action to dispatch into the table, which then maps the widget uids into the entity id that it belongs to (and from there if needed an action to the rest of the app), I'll double check.

But I'll keep an eye for an easier solution when migrating to buttons 👍🏽